```
/home/collabora/jenkins/workspace/github_linux_co-25.04_lokit/sc/source/ui/view/gridwin.cxx: In member function 'void ScGridWindow::UpdateDragRectOverlay()':
/home/collabora/jenkins/workspace/github_linux_co-25.04_lokit/sc/source/ui/view/gridwin.cxx:7289:49: error: 'comphelper::COKit' has not been declared
 7289 |         tools::Long nLayoutSign = (!comphelper::COKit::isActive() && bLayoutRTL) ? -1 : 1;
      |                                                 ^~~~~
/home/collabora/jenkins/workspace/github_linux_co-25.04_lokit/sc/source/ui/view/gridwin.cxx:7326:41: error: 'comphelper::COKit' has not been declared
 7326 |         if ( bLayoutRTL && !comphelper::COKit::isActive() )
      |                                         ^~~~~
make[1]: *** [/home/collabora/jenkins/workspace/github_linux_co-25.04_lokit/solenv/gbuild/LinkTarget.mk:339: /home/collabora/jenkins/workspace/github_linux_co-25.04_lokit/workdir/CxxObject/sc/source/ui/view/gridwin.o] Error 1
```

--- a/browser/src/control/Control.JSDialogBuilder.js
+++ b/browser/src/control/Control.JSDialogBuilder.js
@@ -1672,6 +1672,17 @@ window.L.Control.JSDialogBuilder = window.L.Control.extend({
 							builder.map,
 						);
 					}
+
+					// Swap tooltip text when activeTooltip is set
+					if (data.activeTooltip) {
+						if (isOn) {
+							div._wasActiveTooltip = true;
+							div.setAttribute('data-cooltip', data.activeTooltip);
+						} else if (div._wasActiveTooltip) {
+							div._wasActiveTooltip = false;
+							div.setAttribute('data-cooltip', tooltip);
+						}
+					}
 				};
 
 				updateFunction();

--- a/browser/src/control/Control.NotebookbarCalc.js
+++ b/browser/src/control/Control.NotebookbarCalc.js
@@ -443,6 +443,7 @@ window.L.Control.NotebookbarCalc = window.L.Control.NotebookbarWriter.extend({
 									'type': 'toolitem',
 									'text': _UNO('.uno:FormatPaintbrush'),
 									'tooltip': _('Clone Formatting (double click to keep active)'),
+									'activeTooltip': _('Clone Formatting is active (click again or press Esc to exit)'),
 									'command': '.uno:FormatPaintbrush',
 									'doubleClickCommand': '.uno:FormatPaintbrush',
 									'doubleClickCommandArgs': { PersistentCopy: { type: 'boolean', value: true } },

--- a/browser/src/control/Control.NotebookbarDraw.js
+++ b/browser/src/control/Control.NotebookbarDraw.js
@@ -638,6 +638,7 @@ window.L.Control.NotebookbarDraw = window.L.Control.NotebookbarImpress.extend({
 										'type': 'toolitem',
 										'text': _UNO('.uno:FormatPaintbrush'),
 										'tooltip': _('Clone Formatting (double click to keep active)'),
+										'activeTooltip': _('Clone Formatting is active (click again or press Esc to exit)'),
 										'command': '.uno:FormatPaintbrush',
 										'doubleClickCommand': '.uno:FormatPaintbrush',
 										'doubleClickCommandArgs': { PersistentCopy: { type: 'boolean', value: true } },

--- a/browser/src/control/Control.NotebookbarImpress.js
+++ b/browser/src/control/Control.NotebookbarImpress.js
@@ -770,6 +770,7 @@ window.L.Control.NotebookbarImpress = window.L.Control.NotebookbarWriter.extend(
 										'type': 'toolitem',
 										'text': _UNO('.uno:FormatPaintbrush'),
 										'tooltip': _('Clone Formatting (double click to keep active)'),
+										'activeTooltip': _('Clone Formatting is active (click again or press Esc to exit)'),
 										'command': '.uno:FormatPaintbrush',
 										'doubleClickCommand': '.uno:FormatPaintbrush',
 										'doubleClickCommandArgs': { PersistentCopy: { type: 'boolean', value: true } },

--- a/browser/src/control/Control.NotebookbarWriter.js
+++ b/browser/src/control/Control.NotebookbarWriter.js
@@ -632,6 +632,7 @@ window.L.Control.NotebookbarWriter = window.L.Control.Notebookbar.extend({
 									'type': 'toolitem',
 									'text': _UNO('.uno:FormatPaintbrush'),
 									'tooltip': _('Clone Formatting (double click to keep active)'),
+									'activeTooltip': _('Clone Formatting is active (click again or press Esc to exit)'),
 									'command': '.uno:FormatPaintbrush',
 									'doubleClickCommand': '.uno:FormatPaintbrush',
 									'doubleClickCommandArgs': { PersistentCopy: { type: 'boolean', value: true } },

--- a/browser/src/control/Control.TopToolbar.js
+++ b/browser/src/control/Control.TopToolbar.js
@@ -118,7 +118,7 @@ class TopToolbar extends JSDialog.Toolbar {
 			{type: 'toolitem',  id: 'redo', text: _UNO('.uno:Redo'), command: '.uno:Redo', mobile: false, icon: 'compact_redo.svg'}
 		];
 		var fontGroup = [
-			{type: 'toolitem',  id: 'formatpaintbrush', text: _UNO('.uno:FormatPaintbrush'), tooltip: _('Clone Formatting (double click to keep active)'), command: '.uno:FormatPaintbrush', doubleClickCommand: '.uno:FormatPaintbrush', doubleClickCommandArgs: { PersistentCopy: { type: 'boolean', value: true } }, mobile: false, icon: 'compact_formatpaintbrush.svg'},
+			{type: 'toolitem',  id: 'formatpaintbrush', text: _UNO('.uno:FormatPaintbrush'), tooltip: _('Clone Formatting (double click to keep active)'), activeTooltip: _('Clone Formatting is active (click again or press Esc to exit)'), command: '.uno:FormatPaintbrush', doubleClickCommand: '.uno:FormatPaintbrush', doubleClickCommandArgs: { PersistentCopy: { type: 'boolean', value: true } }, mobile: false, icon: 'compact_formatpaintbrush.svg'},
 			{type: 'toolitem',  id: 'reset', text: _UNO('.uno:ResetAttributes', 'text'), visible: false, command: '.uno:ResetAttributes', mobile: false, icon: 'compact_setdefault.svg'},
 			{type: 'toolitem',  id: 'resetimpress', class: 'unoResetAttributes', text: _UNO('.uno:SetDefault', 'presentation', 'true'), visible: false, command: '.uno:SetDefault', mobile: false, icon: 'compact_setdefault.svg'},
 			{type: 'separator', orientation: 'vertical', id: 'breakreset', invisible: true, mobile: false, tablet: false,},

--- a/browser/src/control/Control.UIManager.ts
+++ b/browser/src/control/Control.UIManager.ts
@@ -122,6 +122,36 @@ class UIManager extends window.L.Control {
 	// UI initialization
 
 	/**
+	 * Shows a tooltip attached to an element and auto-hides it after a timeout.
+	 */
+	showTimedTooltip(element: HTMLElement, text: string, timeMs: number): void {
+		if (!element || !this.map.tooltip)
+			return;
+
+		this.map.tooltip.show(element, text);
+		clearTimeout(this.map._timedTooltipTimeout);
+		this.map._timedTooltipTimeout = setTimeout(() => {
+			if (this.map.tooltip && this.map.tooltip._current === element) {
+				this.map.tooltip.hide();
+			}
+		}, timeMs);
+	}
+
+	/**
+	 * Shows a timed tooltip on an element and optionally plays the attention animation.
+	 */
+	showAttention(element: HTMLElement, text: string, animate: boolean, timeMs: number = 3000): void {
+		this.showTimedTooltip(element, text, timeMs);
+
+		if (animate && !element.classList.contains('attention')) {
+			element.classList.add('attention');
+			element.addEventListener('animationend', function () {
+				element.classList.remove('attention');
+			}, { once: true });
+		}
+	}
+
+	/**
 	 * Returns the current UI mode ("notebookbar" or "classic").
 	 */
 	getCurrentMode(): UIMode {

--- a/browser/src/map/handler/Map.Keyboard.js
+++ b/browser/src/map/handler/Map.Keyboard.js
@@ -7,7 +7,7 @@
  * at TextInput.
  */
 
-/* global app UNOKey TileManager */
+/* global app UNOKey TileManager _ */
 
 window.L.Map.mergeOptions({
 	keyboard: true,
@@ -645,6 +645,13 @@ window.L.Map.Keyboard = window.L.Handler.extend({
 			}
 			else if (key in this._zoomKeys) {
 				map.setZoom(map.getZoom() + (ev.shiftKey ? 3 : 1) * this._zoomKeys[key], null, true /* animate? */);
+			}
+			else if (ev.key && ev.key.length === 1 && !ev.ctrlKey && !ev.altKey && !map.isEditMode()) {
+				let permissionMode = map.uiManager && map.uiManager.permissionViewMode;
+				let viewModeBtn = permissionMode && (permissionMode.viewModeDropdown || permissionMode.viewModeContainer);
+				if (viewModeBtn && map.uiManager && map.uiManager.showAttention) {
+					map.uiManager.showAttention(viewModeBtn, _('You are currently in View mode'), true, 5000);
+				}
 			}
 		}
 

